### PR TITLE
Fix `selector-pseudo-class-allowed-list` reported ranges

### PR DESCRIPTION
**Pull request by ryo-manba**
> Which issue, if any, is this issue related to?

See https://github.com/stylelint/stylelint/issues/7904

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: lib/rules/selector-pseudo-class-allowed-list/index.cjs [hunk 1]
```diff
@@ -63,12 +66,13 @@ const rule = (primary) => {
 				}
 
 				report({
-					word: value,
```
Comment (ryo-manba) [inline lib/rules/selector-pseudo-class-allowed-list/index.cjs:66]: See: https://github.com/stylelint/stylelint/pull/8045#discussion_r1800201009

File: lib/rules/selector-pseudo-class-allowed-list/__tests__/index.mjs [hunk 2]
```diff
@@ -49,6 +51,16 @@ testRule({
...
 		},
+		{
+			code: stripIndent`
+				/* stylelint-disable-next-line selector-pseudo-class-allowed-list */
+				:HOVER, /* stylelint-disable-next-line selector-pseudo-class-allowed-list */
+				:HOVER,
+				/* stylelint-disable-next-line selector-pseudo-class-allowed-list */
+				:HOVER
+				{}
+			`,
```
Comment (romainmenke) [inline lib/rules/selector-pseudo-class-allowed-list/__tests__/index.mjs:62]: Maybe better to use `:focus` in the tests.

That `:hover` and `:HOVER` are treated differently in this rule is peculiar and not what we want to test in this case is:)

**Review by romainmenke — COMMENTED** [review]
One suggestion, otherwise looks good!

File: lib/rules/selector-pseudo-class-allowed-list/__tests__/index.mjs (hunk 2 above)
Comment (ryo-manba) [reply lib/rules/selector-pseudo-class-allowed-list/__tests__/index.mjs]: Ah, that sounds good. I've fixed it.

[chore: rename :HOVER to :focus](https://github.com/stylelint/stylelint/pull/8061/commits/7de08c98903b032bcfad92e5e1566b8423dab9c0)

**Review by romainmenke — APPROVED** [review]
LGTM!

Thank you for the speedy fix 🙇

**Pull request merged**